src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = {};
+  const dateValues = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    if (year < 30) {
+      dateParts['YYYY'] = `20${dateParts['YY']}`;
+    } else {
+      dateParts['YYYY'] = `19${dateParts['YY']}`;
+    }
+  }
```

Comment:
Consider refactoring the conditional block for converting 'YY' to 'YYYY' to a separate function. This would improve readability and maintainability of the code by separating concerns and making the `formatDate` function more focused on the formatting logic itself.